internal/cortex/frontend/transport/handler.go

```diff
@@ -122,14 +153,54 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	r.Body = http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize)
 	r.Body = io.NopCloser(io.TeeReader(r.Body, &buf))
 
+	if f.lru != nil {
+		// Store query expression
+		queryExpressionNormalized = f.regex.ReplaceAllString(r.URL.Query().Get("query"), " ")
+
+		// Time range length for queries, if either of "start" or "end" are not present, return 0
+		getQueryRangeSeconds := func() int {
+			start, err := strconv.Atoi(r.URL.Query().Get("start"))
+			if err != nil {
+				return 0
+			}
+			end, err := strconv.Atoi(r.URL.Query().Get("end"))
+			if err != nil {
+				return 0
+			}
+			return end - start
+		}
+
+		//Store query time range length
+		queryExpressionRangeLength = getQueryRangeSeconds()
+
+		// Check if query in cache and whether value exceeds time range length
+		if value, ok := f.lru.Get(queryExpressionNormalized); ok && value.(int) >= queryExpressionRangeLength {
```

Comment:
shall we have metrics to track this cache hit rate? fyi @hczhu-db

Reply:
Added metrics for total hits and total queries